Frameworks/UIKit/UILabel.mm

```diff
@@ -143,38 +76,38 @@ - (void)_adjustFontSizeToFit {
             }
         }
 
-        StrongId<UIFont> _targetFont = [self _findMaxFontSizeToFit:rect
-                                                              Text:_text
-                                                              Font:_font
-                                                     NumberOfLines:_numberOfLines
-                                                   MinimumFontSize:minimumFontSize
-                                                  StartingFontSize:[_font pointSize]];
-        if (_targetFont != nil) {
-            // found a font that can be adjusted to fit, otherwise, do nothing
-            _font = _targetFont;
-        }
+        // findMaxFontSizeToFit return maximnum Font Size which should not exceed
+        // original fontSize, thus never scale up. use original font size as upper bound
+        // if not found, return nil
+        return XamlUtilities::FindMaxFontSizeToFit(rect, _text, _font, _numberOfLines, minimumFontSize, _originalFontSize);
     }
 
-    [self _updateXamlElement];
+    return nil;
 }
 
-- (void)_updateXamlElement {
+- (void)_updateXamlElementWithAdjustedFont:(UIFont*)adjustedFont {
```

Comment:
already seen some pretty good perf gain after addressing #1497.

---
In reply to: [105066550](https://github.com/Microsoft/WinObjC/pull/2178#discussion_r105066550) [](ancestors = 105066550,105066411,105030923)